pkg/operator/validations.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2025 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"github.com/blang/semver/v4"
+	"github.com/prometheus/common/model"
+)
+
+// ValidationSchemeForPrometheus returns the appropriate validation scheme based on Prometheus version.
+func ValidationSchemeForPrometheus(version semver.Version) model.ValidationScheme {
+	if version.GTE(semver.MustParse("3.0.0")) {
+		return model.UTF8Validation
+	}
+	return model.LegacyValidation
+}
+
+// ValidationSchemeForThanos returns the appropriate validation scheme based on Thanos version.
+func ValidationSchemeForThanos(version semver.Version) model.ValidationScheme {
+	if version.GTE(semver.MustParse("0.39.0")) {
```

Comment:
https://github.com/thanos-io/thanos/commit/0414eef64d6a3897ae4c09ac2fcb0aa1c1fb9cd1 shows that it was shipped starting with v0.38